bin/geninfo

```diff
@@ -491,6 +504,16 @@ if ($intermediate) {
 		     "intermediate format - ignoring\n");
 		$opt_derive_func_data = 0;
 	}
+	if (($br_coverage == 2) && !$gcov_caps->{'json-format'}) {
+		warn("WARNING: excluding exception branches is not compatible with ".
+		     "text intermediate format - ignoring\n");
+	}
+}
+
+if(($br_coverage == 2) && ($gcov_version < $GCOV_VERSION_3_3_0))
+{
+	warn("WARNING: excluding exception branches is not compatible with ".
+	     "gcov versions older than 3.3.0 - ignoring\n");
 }
```

Comment:
This is in a separate block from the previous check because the previous check is only done to define the behavior of "intermediate=auto". Any way, both are logically exclusive because it cannot happen that the intermediate format is "text" if GCOV version is less than 3.3 intermediate.